packages/mco/components/disaster-recovery/applications/failover-and-relocate-modal/modal-body.tsx

```diff
@@ -0,0 +1,319 @@
+import { ResourceIcon } from '@odf/shared/resource-link/resource-link';
+import { getName } from '@odf/shared/selectors';
+import { ApplicationKind } from '@odf/shared/types';
+import { referenceForModel } from '@odf/shared/utils';
+import {
+  useK8sWatchResources,
+  ObjectReference,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { TFunction } from 'i18next';
+import { Flex, FlexItem, Alert, AlertVariant } from '@patternfly/react-core';
+import { DR_SECHEDULER_NAME } from '../../../../constants';
+import {
+  ACMPlacementRuleModel,
+  ACMSubscriptionModel,
+  DRPlacementControlModel,
+} from '../../../../models';
+import {
+  ACMSubscriptionKind,
+  ACMPlacementRuleKind,
+  DRPlacementControlKind,
+} from '../../../../types';
+import {
+  matchApplicationToSubscription,
+  isPlacementRuleModel,
+} from '../../../../utils';
+import { DRPolicySelector } from './dr-policy-selector';
+import { PeerStatus } from './peer-status';
+import {
+  FailoverAndRelocateState,
+  FailoverAndRelocateAction,
+  FailoverAndRelocateType,
+  ACMToDRState,
+  ACTION_TYPE,
+  ErrorMessage,
+} from './reducer';
+import { SubscriptionGroupSelector } from './subscription-group-selector';
+import { TargetClusterSelector } from './target-cluster-selector';
+import './modal-body.scss';
+
+export const findErrorMessage = (
+  errorMessage: ErrorMessage,
+  includeUpdateError: boolean = true
+) =>
+  [
+    errorMessage.acmToDRStateErrorMessage,
+    errorMessage.drPolicyErrorMessage,
+    errorMessage.targetClusterErrorMessage,
+    errorMessage.subscriptionGroupErrorMessage,
+    errorMessage.peerStatusErrorMessage,
+    includeUpdateError && errorMessage.updateErrorMessage,
+  ]
+    .filter(Boolean)
+    .find((errorMessage) => errorMessage);
+
+const isObjectRefMatching = (
+  objectRef: ObjectReference,
+  objectRefList: string[]
+) => objectRefList.includes(objectRef?.name);
+
+const filterDRPlacementRules = (
+  placementRules: ACMPlacementRuleKind[]
+): string[] =>
+  placementRules?.reduce(
+    (arr, placementRule) =>
+      placementRule?.spec?.schedulerName === DR_SECHEDULER_NAME
+        ? [...arr, getName(placementRule)]
+        : arr,
+    []
+  );
+
+const filterDRSubscriptions = (
+  application: ApplicationKind,
+  subscriptions: ACMSubscriptionKind[],
+  drPlacementRules: string[]
+): SubcsriptionMap =>
+  subscriptions?.reduce((arr, subscription) => {
+    const plsRuleName = subscription?.spec?.placement?.placementRef?.name;
+    const subsMap =
+      isPlacementRuleModel(subscription) &&
+      isObjectRefMatching(
+        subscription?.spec?.placement?.placementRef,
+        drPlacementRules
+      ) &&
+      matchApplicationToSubscription(subscription, application)
+        ? {
+            ...arr,
+            [plsRuleName]:
+              plsRuleName in arr
+                ? [...arr[plsRuleName], getName(subscription)]
+                : [getName(subscription)],
+          }
+        : arr;
+    return subsMap;
```

Comment:
nits: `arr` signify that it is an array
```suggestion
  subscriptions?.reduce((obj, subscription) => {
    const plsRuleName = subscription?.spec?.placement?.placementRef?.name;
    const subsMap =
      isPlacementRuleModel(subscription) &&
      isObjectRefMatching(
        subscription?.spec?.placement?.placementRef,
        drPlacementRules
      ) &&
      matchApplicationToSubscription(subscription, application)
        ? {
            ...obj,
            [plsRuleName]:
              plsRuleName in arr
                ? [...arr[plsRuleName], getName(subscription)]
                : [getName(subscription)],
          }
        : obj;
    return subsMap;
```